ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java

```diff
@@ -436,8 +456,16 @@ public void setTextAlign(@Nullable String textAlign) {
 
   @ReactProp(name = ViewProps.FONT_SIZE, defaultFloat = UNSET)
   public void setFontSize(float fontSize) {
-    if (fontSize != UNSET) {
+    mFontSizeInput = fontSize;
+
+    if (fontSize == UNSET) {
+      fontSize = ViewDefaults.FONT_SIZE_SP;
+    }
```

Comment:
When `mFontSizeInput` is `UNSET` maybe we should set `mFontSize` to `UNSET` similar to how we set `mLineHeight` to `NaN` when `mLineHeightInput` is `UNSET`.

The reasoning is that this PR changes the behavior of lines `145` and `195`. Before your PR, if somebody reverted the font size to its default by passing `null` for the `fontSize` prop, `mFontSize` would become `UNSET`. Now, `setFontSize` never leaves `mFontSize` with the value `UNSET`.

Setting `mFontSize` to `UNSET` when `mFontSizeInput` is `UNSET` would preserve the old behavior of these conditionals.